apps/research-service/src/validations/validators/estimated-area.validator.ts

```diff
@@ -0,0 +1,198 @@
+    const upperBound = EstimatedAreaValidator.UPPER_BOUND_MULTIPLIER * site.hectaresToRestoreGoal;
+    const valid = sumAreaSite >= lowerBound && sumAreaSite <= upperBound;
+    const percentage = (sumAreaSite / site.hectaresToRestoreGoal) * 100;
+
+    return {
+      valid,
+      sumAreaSite: Math.round(sumAreaSite),
+      percentageSite: Math.round(percentage),
+      totalAreaSite: site.hectaresToRestoreGoal,
+      lowerBoundSite: lowerBound,
+      upperBoundSite: upperBound
+    };
+  }
+
+  private async generateAreaDataProject(sitePolygon: SitePolygon): Promise<{
+    valid: boolean;
+    sumAreaProject: number | null;
+    percentageProject: number | null;
+    totalAreaProject: number | null;
+    lowerBoundProject: number | null;
+    upperBoundProject: number | null;
+  }> {
+    const site = sitePolygon.site != null ? sitePolygon.site : await sitePolygon.loadSite();
+    if (site == null || site.projectId == null) {
+      return {
+        valid: false,
+        sumAreaProject: null,
+        percentageProject: null,
+        totalAreaProject: null,
+        lowerBoundProject: null,
+        upperBoundProject: null
+      };
+    }
+
+    const project = await Project.findByPk(site.projectId);
+    if (project == null || project.totalHectaresRestoredGoal == null || project.totalHectaresRestoredGoal <= 0) {
+      return {
+        valid: false,
+        sumAreaProject: null,
+        percentageProject: null,
+        totalAreaProject: project?.totalHectaresRestoredGoal ?? null,
+        lowerBoundProject: null,
+        upperBoundProject: null
+      };
+    }
+
+    const sumAreaProject = await this.calculateProjectAreaSum(project.id);
+    const lowerBound = EstimatedAreaValidator.LOWER_BOUND_MULTIPLIER * project.totalHectaresRestoredGoal;
+    const upperBound = EstimatedAreaValidator.UPPER_BOUND_MULTIPLIER * project.totalHectaresRestoredGoal;
+    const valid = sumAreaProject >= lowerBound && sumAreaProject <= upperBound;
+    const percentage = (sumAreaProject / project.totalHectaresRestoredGoal) * 100;
+
+    return {
+      valid,
+      sumAreaProject: Math.round(sumAreaProject),
+      percentageProject: Math.round(percentage),
+      totalAreaProject: project.totalHectaresRestoredGoal,
+      lowerBoundProject: lowerBound,
+      upperBoundProject: upperBound
+    };
+  }
+
+  private async calculateSiteAreaSum(siteUuid: string): Promise<number> {
+    const result = await SitePolygon.sum("calcArea", {
+      where: { siteUuid, isActive: true }
+    });
+    return result ?? 0;
+  }
+
+  private async calculateProjectAreaSum(projectId: number): Promise<number> {
+    const sites = await Site.findAll({
+      where: { projectId },
+      attributes: ["uuid"]
+    });
+
+    if (sites.length === 0) {
+      return 0;
+    }
+
+    const siteUuids = sites.map(site => site.uuid);
+    const result = await SitePolygon.sum("calcArea", {
+      where: {
+        siteUuid: siteUuids,
```

Comment:
This should be handled with a subquery instead of the `findAll` above. Should this be using only approved sites? If so, there's one already ready for you to use (`approvedUuidsSubquery`). If not, you can add this to `Site` and use it:
```
  static uuidsSubquery(projectId: number) {
    return Subquery.select(Site, "uuid").eq("projectId", projectId).literal;
  }
```